python/pfs/drp/stella/fitPfsFluxReference.py

```diff
@@ -1026,6 +1056,42 @@ def getBadMask(self) -> List[str]:
         badMask.update(["EDGE", "ATMOSPHERE"])
         return list(badMask)
 
+    def downsampleModel(self, model, obsSpectrum) -> PfsSimpleSpectrum:
+        """Downsample a model spectrum so that its resolution will not be
+        unmeaningfully higher than the observed spectrum's.
+
+        Parameters
+        ----------
+        model : `PfsSimpleSpectrum`
+            Model spectrum.
+        obsSpectrum : `PfsSimpleSpectrum`
+            Observed spectrum.
+
+        Returns
+        -------
+        downsampled : `PfsSimpleSpectrum`
+            Downsampled model.
+        """
+        if self.config.modelOversamplingFactor <= 0:
+            # Downsampling is disabled
+            return model
+
+        minWavelength = model.wavelength[0]
+        maxWavelength = model.wavelength[-1]
+        dlambdaModel = np.max(model.wavelength[1:] - model.wavelength[:-1])
+        dlambdaObs = np.min(obsSpectrum.wavelength[1:] - obsSpectrum.wavelength[:-1])
```

Comment:
Might want to use `pfs.drp.stella.interpolate.calculateDispersion`?

Reply:
`calculateDispersion` returns an array, but I want scalars here.